File: api/app/models/transaction.rb
class Transaction < ApplicationRecord
  belongs_to :organization
  belongs_to :source, polymorphic: true, optional: true
  belongs_to :asset, optional: true

  monetize :amount_cents, with_model_currency: :currency

  DIRECTIONS = %w[income expense].freeze
  enum :direction, DIRECTIONS.index_with(&:itself)

  validates :currency, presence: true
  validates :category, presence: true
  validates :date, presence: true
end

File: frontend/src/types/transaction.ts
export type TransactionDirection = "income" | "expense"

export interface Transaction {
  id: string
  direction: TransactionDirection
  category: string
  date: string
  currency: string
  amount_cents: number
  amount_formatted: string
  asset_id: string | null
  source: { type: string; id: string } | null
}
